networkd/network_monitor_thread.ml

```diff
@@ -28,11 +28,9 @@ let monitor_whitelist = ref [
 	"vif"; (* This includes "tap" owing to the use of standardise_name below *)
 ]
 
-let xapi_rpc request =
-	Rpc_client.do_rpc_unix
-		~content_type:(Rpc_client.content_type_of_string "text/xml")
-		~filename:(Filename.concat "/var/lib/xcp" "xapi")
-		~path:"/" request
+let xapi_rpc xml =
+  let open Xmlrpc_client in
+  XMLRPC_protocol.rpc ~srcstr:"v6" ~dststr:"xapi" ~transport:(Unix "/var/xapi/xapi") ~http:(xmlrpc ~version:"1.0" "/") xml
```

Comment:
srcstr should not be `v6`, please use something like `xcp-networkd`

Reply:
Minor: `/var/xapi` is a symlink to `/var/lib/xcp`, they both work but I think the former is deprecated, so we could use the `/var/lib/xcp/xapi` path that networkd used before, but either way should work.